### PR TITLE
Create ElectrumPersonalServer.md

### PR DESCRIPTION
**Pull request by maltokyo**
EPS can be integrated into BTCPay Server using the optional docker fragment.  This is the document explaining how to do this.

### CODE REVIEW

**Comment by maltokyo** [comment]
Please review and approve at the same time as @NicolasDorier or someone else approves the main PR here https://github.com/btcpayserver/btcpayserver-docker/pull/227

File: ElectrumPersonalServer.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+[Electrum Personal Server or EPS](https://github.com/chris-belcher/electrum-personal-server) is a personal version of public Electrum servers like [ElectrumX](https://github.com/btcpayserver/btcpayserver-doc/blob/master/ElectrumX.md).
+
+EPS can be integrated into BTCPay Server using the optional docker fragment [opt-add-electrum-ps.yml](https://github.com/btcpayserver/btcpayserver-docker/blob/master/docker-compose-generator/docker-fragments/opt-add-electrum-ps.yml).  Use EPS when you want to use your own full node (included in BTCPay Server) to verify your own transactions privately when using Electrum Wallet. The biggest difference with public Electrum servers (eg. ElectrumX) is that EPS is for monitoring only your own wallet(s).  The "XPUB" (extended public key) of the wallet you use in Electrum must be shared with EPS in order for it to function all all. Other than this, it functions (from an end user perspective) in the same was as ElectrumX etc.  It is easy to integrate into BTCPay just follow the instructions below.
```
Comment (britttttk) [inline ElectrumPersonalServer.md]: "in the same _way_ as ElectrumX etc." typo

**Comment by britttttk** [comment]
This is really a good doc! Nice job @maltokyo Only thing, I'm surprised you don't add your images to the repo.

**Comment by maltokyo** [comment]
Thanks @britttttk ! Fixed typo. 
What do you mean by "add image"?

**Comment by pavlenex** [comment]
Is this the same document where you just replaced ElectrumX with ElectrumPersonalServer 😄 ?

Images also mention ElectrumX.

**Comment by maltokyo** [comment]
Indeed this is just another option to ElectrumX. So I wasn't going to reinvent the wheel.

**Comment by maltokyo** [comment]
@pavlenex added the link in Summary.md, thanks!  Anything else I may have missed?

**Comment by britttttk** [comment]
> What do you mean by "add image"?

https://github.com/btcpayserver/btcpayserver-doc/tree/master/img

Normally images get added here in PRs but doesn't look to me like yours are.. I still approve the PR.

**Comment by maltokyo** [comment]
Ahh. Ok. Thanks. Actually I didn't know that and added in the only way I knew how (copy paste).

**Comment by pavlenex** [comment]
@maltokyo images still mention elextrumX and not EPS, after that is fixed I assume we're good to go, though I personally am not a fan of having duplicate docs, but we'll figure how to handle that in the future I guess.

**Comment by maltokyo** [comment]
I'm ok to integrate it all into one file. Just let me know what's better.

**Comment by pavlenex** [comment]
Let's keep them separate for now, and see how people use/react to them. If it's confusing we'll tackle it, if not, we'll leave it as is.

**Comment by maltokyo** [comment]
Ok. At the moment I'm a bit short of time on travel etc. So I can do the images later. In meantime @NicolasDorier  also needs to approve the main PR for the fragment ;)

**Comment by pavlenex** [comment]
Alright, once that PR gets merged, and images here altered, we're good to merge this one as well then.

**Comment by maltokyo** [comment]
@pavlenex I have removed the offending images with ElectrumX, they were "nice to have" anyway.  I have explained properly what should be expected with text only.

**Comment by NicolasDorier** [comment]
1. Need to rename the environment variable to `EPS_XPUB` (mentioned in the other PR)
2. In `4. Set your environment variables:`, just document how to **add** electrum-ps, ie:

```bash
export BTCPAYGEN_ADDITIONAL_FRAGMENTS="$BTCPAYGEN_ADDITIONAL_FRAGMENTS;opt-add-electrum-ps"
export EPS_XPUB="xpub...blah"
. btcpay-setup.sh -i
```
(after the three back tilt opening the code quote add `bash` so github show syntax color)

File: ElectrumPersonalServer.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+3. Follow the [normal setup and install of BTCPay Server](https://github.com/btcpayserver/btcpayserver-docker#full-installation-for-technical-users), then after this command `cd btcpayserver-docker`, follow the below instructions instead of those in the link.  If you already have a BTCPay Server running, then just follow from the next step.
+
+4. Set your environment variables:
+```
+export BTCPAY_HOST="YOURHOST.com" && export NBITCOIN_NETWORK="mainnet" && export BTCPAYGEN_CRYPTO1="btc" && export BTCPAYGEN_REVERSEPROXY="nginx" && export BTCPAYGEN_LIGHTNING="lnd" && export LIGHTNING_ALIAS="MY_LN" && export LETSENCRYPT_EMAIL="you@example.com" && BTCPAYGEN_ADDITIONAL_FRAGMENTS="opt-add-electrum-ps;opt-more-memory"
+```
+You can run all of that as one command after you tweak it to your needs.  The main part for our purposes in this guide of course is `BTCPAYGEN_ADDITIONAL_FRAGMENTS="opt-add-electrum-ps"`.  Note: `opt-more-memory` can be removed if you like, but I really recommend it if your system has more than 1GB of RAM/memory that you can assign to BTCPay server, it will speed synching your node and the general performance of EPS up drastically.
```
Comment (NicolasDorier) [inline ElectrumPersonalServer.md]: You can remove this whole paragraph

**Comment by maltokyo** [comment]
All done, this is good to go for me unless someone has some other feedback.  Thanks!

File: ElectrumPersonalServer.md (hunk 2 above)
Comment (maltokyo) [reply ElectrumPersonalServer.md]: Done.  
added bash like this 

``` bash

Is it what you mean?  Please fix if it is not right, I don't get it..

**Comment by NicolasDorier** [comment]
Awesome documentation:

`Protip - optionally perform the above steps directly in Electrum Wallet config file before even opening the wallet GUI:`

I think this should be pushed before the GUI way of doing it.
The problem with doing it with the GUI is that you already leaked your addresses to third party.

**Comment by NicolasDorier** [comment]
Ok I merged the docker part, try to recommend people to use this via the config file for the reasons I explained, then let's merge this doc.

**Comment by maltokyo** [comment]
@NicolasDorier I have updated the documentation as you have suggested.
@pavlenex @britttttk please merge when ok for you.

**Pull request merged**